### PR TITLE
Multisensor plugin

### PR DESCRIPTION
**Pull request by ni-cc**
Plugin for the TI SensorTag

### CODE REVIEW

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+
+void DevicePluginMultiSensor::deviceRemoved(Device *device)
+{
+    if (!m_tags.values().contains(device))
+        return;
+
+    auto tag= m_tags.key(device);
+
+    m_tags.take(tag);
```
Comment (t-mon) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp]: `tag->deleteLater();` or `delete tag;` is missing

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.json [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                            "index" : 0
+                        },
+                        {
+                            "name": "mac address",
+                            "type": "QString",
+                            "index": 1,
+                            "inputType": "MacAddress"
+                        }
+                    ],
+                    "stateTypes": [
```
Comment (t-mon) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.json]: Maybe a bool state like `connected` or `available` would be useful if the sensor tag is not available. That would also be the `criticalState` for the deviceClass.

**Comment by ni-cc** [comment]
tag is stored in m_tags as a QSharedPointer, which manages its memories, and frees it when the last reference is destroyed (in line 126)

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.json [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                        {
+                            "id": "645633ad-77d4-45b2-8be8-d6ca7a12eb7a",
+                            "idName": "pressure",
+                            "name": "pressure",
+                            "index": 0,
+                            "type": "double",
+                            "unit": "HectoPascal",
+                            "defaultValue": 0
+                        }
+                    ]
```
Comment (t-mon) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.json]: Is there a possibility to add the buttons and create an `Event` once they are pressed?

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    if (device->deviceClassId() == sensortagDeviceClassId) {
+        auto address = QBluetoothAddress(device->paramValue("mac address").toString());
+        auto name = device->paramValue("name").toString();
+        auto deviceInfo = QBluetoothDeviceInfo(address, name, 0);
+
+        QSharedPointer<SensorTag> tag{new SensorTag(deviceInfo, QLowEnergyController::PublicAddress, this)};
+        connect(tag.data(), &SensorTag::valueChanged, this, &DevicePluginMultiSensor::updateValue);
+        m_tags.insert(tag, device);
+
+        tag->connectDevice();
```
Comment (t-mon) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp]: There is currently no possibility to reconnect the sensor if the connection gets lost without restarting the whole server. You could use `guhTimer()` for automatic reconnect mechanism the sensor or create a `reconnect` action.

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp (hunk 1 above)
Comment (ni-cc) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp]: tag is stored in m_tags as a QSharedPointer, which manages its memory, and frees it when the last reference is destroyed (in line 126)

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+        return DeviceManager::DeviceSetupStatusSuccess;
```
Comment (t-mon) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp]: You could verify if the connected bluetooth device is realy the sensortag and not something else. One possibility would be: `return DeviceManager::DeviceSetupStatusAsync` and `emit deviceSetupFinished` once the service discovery is done.

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp (hunk 4 above)
Comment (ni-cc) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.cpp]: what would be preferable?

**Comment by t-mon** [comment]
Jenkins, ok to test.

**Comment by t-mon** [comment]
Jenkins, retest this please.

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.json (hunk 2 above)
Comment (ni-cc) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.json]: done - 956d0ba

File: plugins/deviceplugins/multisensor/devicepluginmultisensor.json (hunk 3 above)
Comment (ni-cc) [inline plugins/deviceplugins/multisensor/devicepluginmultisensor.json]: done - 845a602

**Comment by t-mon** [comment]
Jenkins, retest this please.

**Pull request merged**